src/libPMacc/include/particles/traits/FilterByRNG.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2016 Marco Garten, Heiko Burau
+ *
+ * This file is part of libPMacc.
+ *
+ * libPMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libPMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libPMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "traits/UsesRNG.hpp"
+#include <boost/mpl/copy_if.hpp>
+#include <boost/mpl/placeholders.hpp>
+
+namespace PMacc
+{
+namespace particles
+{
+namespace traits
+{
+
+/** Return a new sequence of particle species that require random number
+ *  generation for their ionization.
+ *
+ * @tparam T_MPLSeq sequence of particle species
+ *
+ * \TODO rather pass a sequence of objects, like ionization modules
+ */
+template<typename T_MPLSeq>
+struct FilterByRNG
+{
+    typedef T_MPLSeq MPLSeq;
+
+    template<typename T_Species>
```

Comment:
@psychocoderHPC 
As discussed offline:
You'd rather have a generic object passed here, right?
So that the class doesn't know anything about ionization and just checks for the requirement of an RNG. That would mean that we need a trait "ExtractFlag" which returns a list of flags and not a list of species with certain flags. Such a trait sounds fairly useful to me, though.